meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc

```diff
@@ -338,6 +338,7 @@ PREFERRED_PROVIDER_u-boot-default-script:mx8mm-generic-bsp = "u-boot-base-scr"
 PREFERRED_PROVIDER_u-boot-default-script:sota:mx8mm-generic-bsp = "u-boot-ostree-scr-fit"
 KERNEL_IMAGETYPE:sota:mx8mm-generic-bsp = "fitImage"
 KERNEL_CLASSES:sota:mx8mm-generic-bsp = " kernel-lmp-fitimage "
+DISTRO_FEATURES:remove:mx8mm-nxp-bsp = "vulkan"
```

Comment:
Hm, what is actually trying to build/bring vulkan on imx8mm?

DISTRO_FEATURES shouldn't cause the build to fail necessarily, as it is just a generic option for the distribution to support a functionality, and the bsp should be the one then deciding if there is a matching support on the driver level.

A better use at the bsp level is to leverage COMBINED_FEATURES instead, I would say.
```
COMBINED_FEATURES = "${@oe.utils.set_intersect('DISTRO_FEATURES', 'MACHINE_FEATURES', d)}"
COMBINED_FEATURES[vardeps] += "DISTRO_FEATURES MACHINE_FEATURES"
```

Reply:
> Hm, what is actually trying to build/bring vulkan on imx8mm?

lmp-xwayland distro has vulkan enabled in DISTRO_FEATURES 

```
DISTRO_FEATURES:append = " x11 wayland opengl vulkan"
```

https://github.com/foundriesio/meta-lmp/blob/main/meta-lmp-base/conf/distro/lmp-xwayland.conf


> DISTRO_FEATURES shouldn't cause the build to fail necessarily, as it is just a generic option for the distribution to support a functionality, and the bsp should be the one then deciding if there is a matching support on the driver level.

I agree that this should be handled in the bsp layer but is not the case in meta-freescale

Reply:
It seems that this is what is causing the failure:
```
meta-freescale/recipes-graphics/vulkan/vulkan-loader_%.bbappend:RRECOMMENDS:${PN}:append:imxgpu = " libvulkan-imx"
```

If possible I would prefer fixing meta-freescale instead.

Reply:
As the bsp doesn't support vulkan on this machine (i.e: libvulkan-imx is not provided) the only solution I can see is using the vulkan cpu backend Lavapipe provided by mesa to continue using vulkan DISTRO_FEATURES

Reply:
please drop my last comment and I will try to fix the issue in meta-freescale